### DISCUSSION TITLE
PHP 8.1 support for composer v1 series?

### DISCUSSION THREAD

**Discussion by boesing** [composer/composer#10131] [General]
Hey there,

just wondering if there are plans to support PHP 8.1 on composer v1 in any way.
Depending on that, plugin developers might probably drop support for v1 if they start providing support for PHP 8.1.

No demand here - just asking 🤙🏼

**Comment by Seldaek** [comment]
Probably not.. I mean it most likely works with deprecation warnings I assume, but we won't bother running CI on it, it's too messy. People using PHP 8.1 really ought to be using Composer 2.

**Reply by stof** [reply]
being deprecation-free on PHP 8.1 probably requires upgrading symfony/console anyway (the 2.8 branch stopped receiving patches long before the work on supporting PHP 8.1 stricter handling of `null` in core functions). So this would definitely disqualify Composer 1.x (as bumping symfony/console will be done only in Composer 2.2 or 2.3, i.e. the version bumping the min PHP version).
